### PR TITLE
Allow user to ignore some requests

### PR DESCRIPTION
**Pull request by complikator**
You can add `deltafetch_ignore` to your request metadata if you don't want to treat request as visited.

### CODE REVIEW

File: scrapy_deltafetch/middleware.py [hunk 1]
```diff
@@ -92,3 +92,6 @@ def _get_key(self, request):
         key = request.meta.get('deltafetch_key') or request_fingerprint(request)
         # request_fingerprint() returns `hashlib.sha1().hexdigest()`, is a string
         return to_bytes(key)
+
+    def _is_ignored(self, request):
+        return request.meta.get('deltafetch_ignore') is not None
```
Comment (dolohow) [inline scrapy_deltafetch/middleware.py:97]:
```python
request.meta.get('deltafetch_ignore', False)
```

File: tests/test_deltafetch.py [hunk 2]
```diff
@@ -177,11 +177,16 @@ def test_process_spider_output(self):
             Request('http://url', meta={'deltafetch_key': 'key1'}),
 
             # 'test_key_1' is already in the test db --> it should be skipped
-            Request('http://url1', meta={'deltafetch_key': 'test_key_1'})
+            Request('http://url1', meta={'deltafetch_key': 'test_key_1'}),
```
Comment (dolohow) [inline tests/test_deltafetch.py:180]: put this stuff into new test function

**Comment by pawelmhm** [comment]
Looks good, thanks @complikator  I'll review and check if we can add it to next release coming

**Comment by pawelmhm** [comment]
I added docs and changed setting name, adjusted tests and merged your work here #44 

Thanks very much for this contribution!

**Pull request closed without merging**